de.metas.acct.base/src/main/sql/postgresql/ddl/functions/AccountSheetReport.sql

```diff
@@ -0,0 +1,235 @@
+
+    v_time := logDebug('created empty temporary table', v_time);
+
+
+    --
+    -- insert into the temp table the beginningBalance for all the available accounts which have the balance != 0
+    -- noinspection SqlInsertValues
+    WITH filteredElementValues AS
+             (
+                 SELECT ev.c_elementvalue_id
+                 FROM c_elementvalue ev
+                 WHERE TRUE
+                   AND ev.accounttype = 'A'
+                   AND (p_account_id IS NULL OR ev.c_elementvalue_id = p_account_id)
+                 ORDER BY ev.c_elementvalue_id
+             ),
+         previousBalances AS
+             (
+                 SELECT (de_metas_acct.acctBalanceToDate(ev.c_elementvalue_id, p_c_acctschema_id, (p_dateFrom - INTERVAL '1 day')::date, p_ad_org_id)::de_metas_acct.BalanceAmt).Balance previousDayBalance,
+                        ev.c_elementvalue_id
+                 FROM filteredElementValues ev
+             ),
+         nonZeroPreviousBalances AS
+             (
+                 SELECT *
+                 FROM previousBalances
+                 WHERE previousDayBalance != 0
+             )
+    INSERT
+    INTO tbpFilteredFactAcct (beginningBalance, endingBalance, lineType, account_id)
+    SELECT nonZero.previousDayBalance,
+           nonZero.previousDayBalance,
+           LINE_TYPE_BEGINNINGBALANCE,
+           nonZero.c_elementvalue_id
+    FROM nonZeroPreviousBalances nonZero;
+
+    SELECT count(1) FROM tbpFilteredFactAcct INTO v_temp;
+    v_time := logDebug('inserted beginningBalance: ' || v_temp || ' records', v_time);
+
+
+    --
+    -- insert the fact_acct rows into the table
+    WITH filteredFactAcct AS
+             (
+                 SELECT fa.*,
+                        tc.c_taxcategory_id,
+                        tbp.beginningBalance::numeric beginningBalance,
+                        tbp.endingBalance::numeric    endingBalance,
+                        LINE_TYPE_TRANSACTION
+                 FROM fact_acct fa
+                          INNER JOIN tbpFilteredFactAcct tbp ON tbp.account_id = fa.account_id --
+                          LEFT JOIN c_tax t ON fa.c_tax_id = t.c_tax_id
+                          LEFT JOIN c_taxcategory tc ON t.c_taxcategory_id = tc.c_taxcategory_id
+                 WHERE TRUE
+                   -- AND ev.accounttype = 'A' -- no longer needed here, since we're joining with the already filtered tbpFilteredFactAcct
+                   AND fa.c_acctschema_id = p_c_acctschema_id
+                   AND (fa.dateacct >= p_dateFrom AND fa.dateacct <= p_dateTo)
+                   AND (p_account_id IS NULL OR fa.account_id = p_account_id)
+                   AND (p_c_activity_id IS NULL OR fa.c_activity_id = p_c_activity_id)
+                   AND (p_c_project_id IS NULL OR fa.c_project_id = p_c_project_id)
+             )
+    INSERT
+    INTO tbpFilteredFactAcct
+    SELECT *
+    FROM filteredFactAcct;
+
+    SELECT count(1) FROM tbpFilteredFactAcct INTO v_temp;
```

Comment:
same, avoid using `select count(1)` for logging purposes

Reply:
fixed